File: WINGS_AI_Bootcamp/Elixir/checkPrime.ex
defmodule PrimeCheck do
  def is_prime(n) when n in [2, 3, 5], do: true
  def is_prime(n) when n <= 1 or rem(n, 2) == 0, do: false

  def is_prime(n) do
    check_prime(n, 3)
  end

  defp check_prime(n, r) when r * r > n, do: true
  defp check_prime(n, r) do
    if rem(n, r) == 0 do
      false
    else
      check_prime(n, r + 2)
    end
  end
end

# Hardcoded test
a = 29

if PrimeCheck.is_prime(a) do
  IO.puts("#{a} is a prime number")
else
  IO.puts("#{a} is not a prime number")
end

File: WINGS_AI_Bootcamp/Haskell/checkPrime.hs
isPrime :: Int -> Bool
isPrime n
  | n `elem` [2, 3, 5] = True
  | n <= 1 || even n    = False
  | otherwise          = checkPrime n 3

checkPrime :: Int -> Int -> Bool
checkPrime n r
  | r * r > n        = True
  | n `mod` r == 0   = False
  | otherwise        = checkPrime n (r + 2)

main :: IO ()
main = do
  let a = 29
  if isPrime a
    then putStrLn (show a ++ " is a prime number")
    else putStrLn (show a ++ " is not a prime number")

